weave

```diff
@@ -1590,23 +1621,30 @@ launch_router() {
         fi
     fi
 
+    # Generate a hash of the password so we can tell if it has changed.
+    # Salt the hash with the peer name to make it even harder to reverse
+    [ -n "$WEAVE_PASSWORD" ] && PASSWORD_HASH="--password-hash=$(echo $PEERNAME$WEAVE_PASSWORD | sha256sum | cut -c1-32)"
+
     # Set WEAVE_DOCKER_ARGS in the environment in order to supply
     # additional parameters, such as resource limits, to docker
     # when launching the weave container.
-    ROUTER_CONTAINER=$(docker run --privileged --net=host -d --name=$CONTAINER_NAME \
+    start_if_necessary router $CONTAINER_NAME $IMAGE \
+        --privileged --net=host \
```

Comment:
we could move the `--privileged --net=host` into `start_if_necessary`, or, rather, a `start_container` function called by `start_if_necessary`.